src/backend/gporca/libgpopt/src/xforms/CXformFactory.cpp

```diff
@@ -374,23 +372,12 @@ CXformFactory::Init()
 		CMemoryPoolManager::GetMemoryPoolMgr()->Destroy(mp);
 		m_pxff = NULL;
```

Comment:
https://github.com/arenadata/gpdb/pull/525#issuecomment-1623073108

Reply:
> does the nulling of static variable (and at other ctors too) is needed

Yes, this is then checked in the assert in the destructor.

Reply:
I agree, that destructor checks it for the `NULL`, but seems in case of failed [`Init`](https://github.com/arenadata/gpdb/blob/1fdfa24975800876cc6cdb568e1690409db958b6/src/backend/gporca/libgpopt/src/xforms/CXformFactory.cpp#L372), GPOS_NEW will raise an exception and this variable won't changed (it was a NULL and doesn't changed)

Reply:
The comment was solved in ADBDEV-3913 - no changes needed https://github.com/arenadata/gpdb/pull/555#discussion_r1257528940